server/main.py

```diff
@@ -218,38 +202,44 @@ async def room(websocket: WebSocket) -> None:
     """
     client = Client(websocket)
     await client.accept()
+
     try:
-        initial_data = ConnectionData(**await client.receive())
+        initial_event = Event(**await client.receive())
     except WebSocketDisconnect:
         return
 
-    room_code = initial_data.data.room_code
-    if initial_data.type == "connect":
-        match initial_data.data.connection_type:
-            case "create":
-                manager.create_room(client, room_code)
-            case "join":
-                try:
-                    manager.join_room(client, room_code)
-                except RoomNotFoundError as e:
-                    # Send off to frontend to handle
-                    await client.send(
-                        {
-                            "type": "error",
-                            "data": {
-                                "message": e.message,
-                            },
-                            "status_code": StatusCode.ROOM_NOT_FOUND,
-                        }
-                    )
-                    await client.close()
-                    return
-            case _:
-                raise NotImplementedError
+    if initial_event.type != EventType.CONNECT:
+        return
+
+    initial_data: ConnectData = initial_event.data
+    print(initial_event, initial_data)
+    room_code = initial_data.room_code
+
+    match initial_data.connection_type:
+        case "create":
+            manager.create_room(client, room_code)
+        case "join":
+            try:
+                manager.join_room(client, room_code)
+            except RoomNotFoundError as e:
+                await client.send(
+                    Event(type=EventType.ERROR, data=ErrorData(message=e.message), status_code=StatusCode.SUCCESS)
+                )
+                await client.close()
+                return
+
+    manager.broadcast(initial_event, room_code)
```

Comment:
Can probably recreate the ConnectionManager.connect event to abstract this away

Reply:
I probably wouldn't leave the `ConnectionManager` the job of handling events.

Reply:
This would only be for connecting the client to the room though, seems fitting it would go in the `ConnectionManager`, no? Unless you had another idea.

Event handling will mostly be between receiving the data and broadcasting in the `while True:` loop. So it would be either in the client, manager, or another class.

Reply:
Fixed in aff01c8f0c7d4c14ef2df3a682e569d16ea1e8ca. But I left the error handling for `RoomNotFoundError` in the `room` endpoint.